### ISSUE TITLE
Question: How to run ArchiveBox under Docker as non-root.

### ISSUE THREAD

**Issue by thprice** [ArchiveBox/ArchiveBox#633]
Actually, I run it on Kubernetes, not plain Docker (or, rather, try to run it).

When trying to initialise the archive (database etc.) using
`archivebox init`
it fails with:
`[!] ArchiveBox should never be run as root!`

When running the container as another user than root, it complains that it either doesn't know the user id or fails with:
`error: failed switching to "archivebox": operation not permitted`

Any ideas?

**Comment by pirate** [comment]
I think this is the same issue as here: https://github.com/ArchiveBox/ArchiveBox/issues/539

You can try some of the workarounds others have posted in that thread, or just subscribe and stay tuned for when a fix is pushed.

This only happens in some specific docker scenarios and I think you were unlucky enough to hit one of them. Hopefully we'll have a fix out soon, but I can't promise it'll be this week.

**Comment by thprice** [comment]
Thanks for your response. It may be the same issue but I'm not sure. #539 is about file system access problems while this here seems to be about errors switching the uid.

No need for a reopen, though. I'll wait for the fix and try again. Maybe, I'll come back then. :-)

**Comment by pirate** [comment]
The fix for UID/GID switching issues is the same fix as the one for #539 (I think), it's actually already on `dev` if you want to try it.

```bash
git clone --recurse-submodules -b dev https://github.com/ArchiveBox/ArchiveBox
cd ArchiveBox
docker build . -t archivebox

docker run -v $PWD:/data -it archivebox ...
```

**Comment by thprice** [comment]
Thanks for the fix. It can confirm that it works (at least for me :-))
